### PR TITLE
feat: add sorted sets fetch by score proto

### PR DESCRIPTION
**Pull request by wandaitzuchen**

### CODE REVIEW

File: proto/cacheclient.proto [hunk 1]
```diff
@@ -455,18 +455,45 @@ message _SortedSetFetchRequest {
...
 
+  // num_results fields
   message _All {}
 
   message _Limit {
     uint32 limit = 1;
   }
 
+  // range fields
+  message _ByIndex {}
```
Comment (gautamomento) [inline proto/cacheclient.proto:466]: Should this also have structures similar to endpoints?

File: proto/cacheclient.proto [hunk 2]
```diff
@@ -455,18 +455,45 @@ message _SortedSetFetchRequest {
...
+
+  message _ByScore {
+    message _Endpoint {
+      message _Score {
+        double score = 1;
+        bool open = 2;
```
Comment (gautamomento) [inline proto/cacheclient.proto:472]: A comment would be useful. I believe this means inclusive of the score or not?

**Review by gautamomento — DISMISSED** [review]
Just some nits but LGTM.

File: proto/cacheclient.proto (hunk 1 above)
Comment (wandaitzuchen) [reply proto/cacheclient.proto:466]: What's already implemented is "ByIndex", and it doesn't require additional arguments.

File: proto/cacheclient.proto (hunk 2 above)
Comment (wandaitzuchen) [reply proto/cacheclient.proto]: Added! thanks.

**Review by gautamomento — APPROVED** [review]

**Pull request merged**